### PR TITLE
Enforce recovery readiness and required CI gates

### PR DESCRIPTION
**Pull request by benhook1013**
## Summary

Converges recovery-readiness, deployment-evidence, observability, and required-CI contracts around one fail-closed pre-v1 boundary.

- Blocks every player-facing production promotion class until durable recovery inventory and linked immutable evidence can be verified.
- Keeps executable waivers disabled until trusted one-time consumption authority exists.
- Defines recovery-controller finalization as the authority for first-live and reopen; checked-in JSON remains post-finalization evidence.
- Aligns backup/recovery observability, runbooks, evidence schemas, and operator-facing implementation status.
- Makes required PR gates resilient to metadata edits and ensures `CodeQL Gate` runs for both protected bases.
- Makes runtime ownership creation idempotent under concurrent first-tick creation.

## What Changed

- Hardened deployment preflight validation, report freshness/consumption, evidence binding, and production fail-closed behavior.
- Expanded canonical recovery and traffic-open evidence contracts without claiming currently unreachable diagnostics execute.
- Replaced backup tick-pause readiness signals with artifact lineage, restore readability, and recovery convergence recordings/alerts.
- Added dependency-free observability validation for required alerts/recordings, expression shape, duplicate definitions, and Markdown YAML snippets.
- Isolated metadata-only workflow runs from required validation contexts while preserving base-retarget execution.
- Added atomic natural-key creation/reload for `runtime_region_status` and concurrent integration proof.

## Validation

- `bash dev-tools/validation/run-locked-gradle.sh check`
- `bash dev-tools/validation/run-locked-gradle.sh :game-session-service:check :tcp-proxy-service:check linkCheck lintMarkdown -PfullCheck`
- `./gradlew :game-session-service:integrationTest --tests 'net.firedevops.firemud.gamesession.repository.RuntimeRegionStatusRepositoryIntegrationTest'`
- `./gradlew :tcp-proxy-service:crossServiceTest --tests 'net.firedevops.firemud.tcpproxy.TelnetGatewayGameSessionAccountCrossServiceIntegrationTest.telnetWhisperDeliversTargetAndObserverViews'`
- `bash dev-tools/tests/preflight-contract.sh`
- `bash dev-tools/tests/overlay-preflight-contract.sh`
- `bash dev-tools/tests/observability-validator-contract.sh`
- `bash dev-tools/tests/pr-required-gate-context-contract.sh`
- `python3 -S dev-tools/observability/validate-observability-contract.py`
- `actionlint`
- `shellcheck dev-tools/tests/observability-validator-contract.sh dev-tools/tests/pr-required-gate-context-contract.sh`
- `git diff --check`

## Known Fail-Closed Gaps

Production recovery-controller reads, complete inventory/evidence dereferencing, and nested participant/confidentiality/hardening/controlled-reopen validation remain explicit implementation gaps. Production promotion and reopen stay blocked until those capabilities exist.

## Summary

- Hardened fail-closed recovery and deployment readiness across production, staging, and hobby environments. Recovery evidence now includes immutable lineage, erasure high-water marks, deployment-event bindings, compatibility results, confidentiality evidence, and controller-authoritative `ready_to_reopen → releasing → finalized` transitions via `continueRecovery(...)`.
- Strengthened `dev-tools/deploy/preflight.py`, overlay validation, credential/sanitization validation, traffic-open evidence generation, and production attestation checks. Waivers and incomplete controller authority now block promotion or reopen.
- Updated recovery, deployment, backup, CI/CD, observability, monitoring, and testing architecture/runbook documentation to reflect the new contracts and implementation gaps.
- Added environment- and deployment-event-scoped evidence paths and schemas, including `deploymentEventId`, staging sanitization artifacts, production recovery compatibility, and finalized traffic-open projections. Hobby backup compliance is explicitly marked incomplete.
- Updated GitHub Actions workflows for metadata-only PR edits, fail-closed required gates, CodeQL enforcement on `main` and `develop`, credential-free PR image smoke builds, trusted image publishing, scoped permissions, and stronger full-smoke result matching.
- Reworked backup observability with environment-scoped convergence metrics, artifact lineage/readability signals, missing-source alerts, 30-day restore-drill freshness, and updated Grafana panels. Added observability contract validation.
- Hardened game-session tick processing: lease-fenced Redis scripts, transactional durable batch/effect mutations, optimistic ownership CAS, idempotent runtime-region initialization, stale-owner rejection, and concurrency integration tests in `game-session-service`.
- Replaced the shell `cloc-report` wrapper with a Python CLI and expanded its contract tests; removed the obsolete `firemud-backup.sh` backup lane.
- Added or expanded contract coverage for workflows, preflight, overlays, observability, external credentials, smoke images, architecture documentation, and runtime image publishing.

## Validation and gaps

Validation covers Gradle/integration and concurrency tests, workflow and shell contracts, observability checks, `actionlint`, `shellcheck`, and whitespace validation. Production promotion and player-facing first-live/reopen remain intentionally blocked because durable production recovery-controller reads, complete inventory/evidence dereferencing, and full nested participant, confidentiality, hardening, and controlled-reopen validation are not yet implemented.

### CODE REVIEW

**Comment by benhook1013** [comment]
@coderabbitai full review

**Comment by benhook1013** [comment]
@coderabbitai full review

**Comment by benhook1013** [comment]
@coderabbitai full review

**Comment by benhook1013** [comment]
@coderabbitai full review

**Comment by benhook1013** [comment]
@coderabbitai full review

**Comment by benhook1013** [comment]
@coderabbitai full review

**Comment by benhook1013** [comment]
@coderabbitai full review

**Comment by benhook1013** [comment]
@coderabbitai full review

**Comment by benhook1013** [comment]
@coderabbitai full review

File: services/game-session-service/src/test/java/integration/net/firedevops/firemud/gamesession/repository/RuntimeRegionStatusRepositoryIntegrationTest.java [hunk 1]
```diff
@@ -0,0 +1,297 @@
...
+    dataSource.setUrl(postgres.getJdbcUrl());
+    dataSource.setUsername(postgres.getUsername());
+    dataSource.setPassword(postgres.getPassword());
+
+    Flyway.configure()
+        .dataSource(dataSource)
+        .locations(
+            "filesystem:" + Path.of("src/main/resources/db/migration").toAbsolutePath().normalize())
+        .load()
+        .migrate();
```
[the remark this answers is not in the document]
Comment (benhook1013) [reply services/game-session-service/src/test/java/integration/net/firedevops/firemud/gamesession/repository/RuntimeRegionStatusRepositoryIntegrationTest.java:51]: Verified but not applied: this integration runtime includes multiple service jars, so `classpath:db/migration` discovers both Game Session and Entity Management `V1__baseline.sql` migrations and Flyway fails with duplicate version 1. The existing service-scoped filesystem location is intentional. The exact `RuntimeRegionStatusRepositoryIntegrationTest` passes after retaining it.

**Comment by benhook1013** [comment]
@coderabbitai full review

**Comment by benhook1013** [comment]
@coderabbitai full review

**Comment by benhook1013** [comment]
@coderabbitai full review

**Comment by benhook1013** [comment]
@coderabbitai full review
